WordPress/gutenberg
# Framework: Pin nvmrc to specific current LTS

Related Slack discussion ([link requires registration](https://make.wordpress.org/chat/)): https://wordpress.slack.com/archives/C02QB2JS7/p1588794429373800

This pull request seeks to update `.nvmrc` to designate the actual current LTS version. This is intended to future-proof builds which occur when checking out an older version of the branch. Notably, this is an issue today when checking out old versions of Gutenberg, where we had used an older version of the [`node-sass` dependency](https://www.npmjs.com/package/node-sass) which only works in e.g. Node 10 or earlier. If, at the time, `.nvmrc` had specified `10` as the Node version of the project, it would not be as much an issue†. With these changes, we can at least try to correct this problem moving forward.

Note that the primary downside here is that we will need to manually update this value over time, as the LTS changes. Based on the [release schedule](https://nodejs.org/en/about/releases/), this should be expected to occur once every year (each October).

**Testing Instructions:**

Running `nvm install` should install Node v12.16.3:

```
nvm install --latest-npm && nvm use
```

**Follow-up tasks:**

† This only really helps if one knows to run `nvm install && nvm use` after checking out an older branch / commit. Ideally we can make this more seamless, or at least more noisy as part of checks prior to any actions we expect to depend on this behavior.

Example ideas:

- `post-checkout` git hook to run `nvm install && nvm use`
- Something similar to [this official documented automation](https://github.com/nvm-sh/nvm/blob/master/README.md#automatically-call-nvm-use) 
- Better integration with [`prebuild` `check-engines` step](https://github.com/WordPress/gutenberg/blob/70a4af60c50926b696787a7f74c7eb58eb31e37a/package.json#L188)
   - Specify `engines` in `package.json`?
   - Support `.nvmrc` in `check-engines`?
